addons/website_sale/static/src/scss/website_sale.scss

```
@@ -336,9 +342,11 @@ $input-border-color: $gray-400;
         }
 
         .oe_product:not(.oe_product_size_stretch) {
-            .o_wsale_product_btn {
-                @include o-position-absolute(auto, auto, calc(100% + #{map-get($spacers, 2)}), map-get($spacers, 2));
-                z-index: 2;
+            &:not(#o_wsale_products_grid.o_wsale_context_thumb_hidden .oe_product) {
```

We tend to use `:where` when we are using `:not` or `:has` selectors to avoid increasing specificity
```suggestion
            &:where(:not(#o_wsale_products_grid.o_wsale_context_thumb_hidden .oe_product)) {
```